extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java

```diff
@@ -0,0 +1,73 @@
+package io.quarkus.smallrye.graphql.runtime;
+
+import java.util.function.Supplier;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.ShutdownContext;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.smallrye.graphql.runtime.spi.QuarkusClassloadingService;
+import io.quarkus.vertx.http.runtime.ThreadLocalHandler;
+import io.smallrye.graphql.cdi.producer.GraphQLProducer;
+import io.smallrye.graphql.schema.model.Schema;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+@Recorder
+public class SmallRyeGraphQLRecorder {
+
+    public void createExecutionService(BeanContainer beanContainer, Class<GraphQLProducer> beanClass, Schema schema) {
+        GraphQLProducer graphQLProducer = beanContainer.instance(beanClass);
```

Comment:
Why do you pass the bean class in like this, when it is hard coded anyway? Why not just use GraphQLProducer.class?